server/src/constants.ts

```diff
@@ -42,7 +42,8 @@ export const defaultSettings: JSONColorTokenSettings = {
 /* A color token can be with/without opacity (last two digits)
  * e.g. without opacity #000000, with opacity #00000050
  */
-export const colorTokenPattern = /#[0-9a-fA-F]{6}([0-9]{2})?/g;
+export const colorTokenPattern = /#([0-9a-fA-F]{6}([0-9a-fA-F]{2})?|[0-9a-fA-F]{3}([0-9a-fA-F]{1})?)([ ;,"]|$)/g;
```

Comment:
Is it necessary to enforce a delimiter that is `[ ;,"]|$` after each token?

It will break color preview for tokens written next to each other like `#aaaaaa#bbbbbb`.

Reply:
well the problem was `#12345` declared as valid colro and and also `#123456789` which thay are not

Reply:
For `#123456789`, interpreting it as `#12345678`, which is a valid full hex code and discard the remaining 9 is an acceptable behavior. I believe it's very rare to not have any delimiter after the color token. But assuming the delimiter is just one of the 5 options is too aggressive here. I feel that `/#([0-9a-fA-F]{6}([0-9a-fA-F]{2})?|[0-9a-fA-F]{3}([0-9a-fA-F]{1})?)/g` can do the job since it will attempt to match a full hex code first and then attempt to match a short hex code if that fails. If you can get an example where this is obviously broken I will accept the delimiter approach.

Reply:
well actually i just removed the `([ ;,"]|$)` from regex and its worked :)